### PR TITLE
[MM-13837] Fixed curly brace not included in link

### PR DESCRIPTION
**Pull request by hahmadia**
**Summary**
Allows curly brace to be at the end of a link without leaving it out.

Before fix:
image

After fix:
image

Ticket Link
[Github Ticket](https://github.com/mattermost/mattermost-server/issues/10774)
[Jira Ticket](https://mattermost.atlassian.net/browse/MM-13837)

### CODE REVIEW

**Comment by hanzei** [comment]
@hmhealey Would you please review this?

File: lib/marked.js [hunk 1]
```diff
@@ -539,7 +539,7 @@ inline.pedantic = merge({}, inline.normal, {
 
 inline.gfm = merge({}, inline.normal, {
   escape: replace(inline.escape)('])', '~|])')(),
-  url: XRegExp.cache('^((?:[A-Za-z][A-Za-z\\d-.+]*:(?:\\/{1,3}|[\\pL\\d%])|www\\d{0,3}[.]|[\\pL\\d.\\-]+[.]\\pL{2,4}\\/)(?:\\[[\\da-f:]+\\]|[^\\s`!()\\[\\]{};:\'",<>?«»“”‘’]|[`!\\[\\]{};:\'",<>?«»“”‘’](?=[^\\s()<>])|\\((?:[^\\s()<>]|(?:\\([^\\s()<>]+\\)))*\\))+)', ''),
+  url: XRegExp.cache('^((?:[A-Za-z][A-Za-z\\d-.+]*:(?:\\/{1,3}|[\\pL\\d%])|www\\d{0,3}[.]|[\\pL\\d.\\-]+[.]\\pL{2,4}\\/)(?:\\[[\\da-f:]+\\]|[^\\s`!()\\[\\]{;:\'",<>?«»“”‘’]|[`!\\[\\]{};:\'",<>?«»“”‘’](?=[^\\s()<>])|\\((?:[^\\s()<>]|(?:\\([^\\s()<>]+\\)))*\\))+)', ''),
```
Comment (hmhealey) [inline lib/marked.js]: Same comment here as in your other PR: https://github.com/mattermost/commonmark.js/pull/8#discussion_r308310331

```suggestion
  url: XRegExp.cache('^((?:[A-Za-z][A-Za-z\\d-.+]*:(?:\\/{1,3}|[\\pL\\d%])|www\\d{0,3}[.]|[\\pL\\d.\\-]+[.]\\pL{2,4}\\/)(?:\\[[\\da-f:]+\\]|[^\\s`!()\\[\\]{;:\'",<>?«»“”‘’]|[`!\\[\\]{;:\'",<>?«»“”‘’](?=[^\\s()<>])|\\((?:[^\\s()<>]|(?:\\([^\\s()<>]+\\)))*\\))+)', ''),
```

**Review by hmhealey — CHANGES_REQUESTED** [review]

File: lib/marked.js (hunk 1 above)
Comment (hahmadia) [reply lib/marked.js]: Resolved! Thanks 😃

**Review by hmhealey — APPROVED** [review]
Thanks!

**Pull request merged**